x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/LookupFromIndexService.java

```diff
@@ -78,9 +79,19 @@ protected TransportRequest transportRequest(LookupFromIndexService.Request reque
     @Override
     protected QueryList queryList(TransportRequest request, SearchExecutionContext context, Block inputBlock, DataType inputDataType) {
         MappedFieldType fieldType = context.getFieldType(request.matchField);
+        validateTypes(request.inputDataType, fieldType);
         return termQueryList(fieldType, context, inputBlock, inputDataType);
     }
 
+    private static void validateTypes(DataType inputDataType, MappedFieldType fieldType) {
```

Comment:
Not so much a bug-fix as early error generation. This could be made less strict going forward, but we thought to start strict and then open later if appropriate.